### DISCUSSION TITLE
Integrating Elsa as Consuer Message Broker

### DISCUSSION THREAD

**Discussion by nurhidayat-agung** [elsa-workflows/elsa-core#5566] [Q&A]
hello elsa,

is posible elsa become consumer message broker like Rabbit MQ ?

**Comment by glime-ai** [comment]
Yes, Elsa can be used as a consumer of messages from a message broker like RabbitMQ. Elsa integrates with MassTransit, which is a message bus for .NET that can use various transport providers, including RabbitMQ. By configuring Elsa to use MassTransit with RabbitMQ, you can have Elsa workflows that are triggered by messages arriving on RabbitMQ queues.

To set up Elsa as a consumer with RabbitMQ, you would typically install the necessary NuGet package for RabbitMQ integration and configure the transport provider within your application's startup configuration. Here's an example of how you might configure Elsa to use MassTransit with RabbitMQ:

```csharp
services.AddElsa(elsa => 
{
    elsa.UseMassTransit(massTransit => 
    {
        massTransit.UseRabbitMq("amqp://guest:guest@localhost:5672/elsa", rabbitMqFeature => 
        {
            rabbitMqFeature.ConfigureRabbitMq = bus => 
            {
                bus.PrefetchCount = 4;
                bus.Durable = true;
                bus.AutoDelete = false;
                bus.ConcurrentMessageLimit = 32;
                // etc.
            };
        });
    });
});
```

This configuration allows Elsa to consume messages from RabbitMQ and trigger workflows based on those messages. You can find more information on how to set up and configure Elsa with MassTransit and RabbitMQ in the Elsa documentation.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
